cmd/kubeadm/app/util/config/masterconfig.go

```diff
@@ -59,17 +59,25 @@ func SetInitDynamicDefaults(cfg *kubeadmapi.MasterConfiguration) error {
 		cfg.KubeProxy.Config.BindAddress = kubeadmapiv1alpha2.DefaultProxyBindAddressv6
 	}
 	// Resolve possible version labels and validate version string
-	err = NormalizeKubernetesVersion(cfg)
-	if err != nil {
+	if err := NormalizeKubernetesVersion(cfg); err != nil {
 		return err
 	}
 
-	if cfg.Token == "" {
-		var err error
-		cfg.Token, err = tokenutil.GenerateToken()
+	// Populate the .Token field with a random value if unset
+	for i, bt := range cfg.BootstrapTokens {
```

Comment:
Might be a silly question, but what are the use cases when `.Token` is unset (assuming that everything passes through the API validation before)?

Reply:
Ok. If we are not allowed to return errors it makes sense. Thanks for the explanation

Reply:
Added that comment to the code